### PR TITLE
meta: add nix tooling

### PR DESCRIPTION
**Pull request by FintanH**
Adds nix tooling to allow people to develop using nix dependencies.

Uses niv to pin nixpkgs, rust-overlay, and niv itself.

This will give Rust tooling that is up to date with rustc-1.64.0.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+, rust-overlay ? pkgs.rust-bin.stable.latest.default
+}:
+  with pkgs;
+  mkShell {
+    name = "build";
+    buildInputs = [
+        # cargo tooling
+        cargo-deny
+        cargo-watch
+        pkgs.rust-bin.nightly."2022-07-01".rustfmt
```
Comment (cloudhead) [inline default.nix:14]: What's up with this?

Comment (FintanH) [reply default.nix:14]: Oh, I can get rid of that. We need nightly in `radicle-git` for formatting -- which I copied this from. I'll remove it here.

**Comment by FintanH** [comment]
@cloudhead you happy to merge this?

**Comment by cloudhead** [comment]
> @cloudhead you happy to merge this?

Happy is a big word :joy: I'm just getting used to the idea of having all these obscure nix files in my repo. Is there a way to like, tuck them away somehow?

**Comment by cloudhead** [comment]
You know like, `.nix/`, and have the shell/default files in there as well? That way it doesn't distract non-nix users?

**Comment by FintanH** [comment]
> You know like, `.nix/`, and have the shell/default files in there as well? That way it doesn't distract non-nix users?

Poor nix users trying to not distract non-nix users :(

I've pushed a change to do that and I can always start the shell with `nix-shell .nix/shell.nix` -- annoying but fair.

**Pull request merged**